### PR TITLE
app/vmselect/promql: properly handle possible negative results caused…

### PR DESCRIPTION
**Pull request by Haleygo**
… by float operations precision error in rollup functions like rate() or increase()

related https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5571

### CODE REVIEW

**Comment by hagen1778** [comment]
Tests are failing(

File: app/vmselect/promql/exec_test.go [hunk 1]
```diff
@@ -8071,7 +8071,7 @@ func TestExecSuccess(t *testing.T) {
 		}}
 		r3 := netstorage.Result{
 			MetricName: metricNameExpected,
-			Values:     []float64{4, 3, 2, 1, 0, -1},
```
Comment (Haleygo) [inline app/vmselect/promql/exec_test.go:8074]: Bump `2000` in test expr to `2200`, since now `removeCounterResets` wouldn't return negative value,

**Comment by Haleygo** [comment]
> Tests are failing(

Sorry, fixed, please check)

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@Haleygo , thanks for the fix!

**Pull request merged**